pulp_container/app/registry_api.py

```diff
@@ -691,32 +667,19 @@ def put(self, request, path, pk=None):
         digest = request.query_params["digest"]
         # Try to see if the client came back after we told it to backoff with the ``Throttled``
         # exception. In that case we answer based on the task state, or make it backoff again.
-        # This mechanism seems to work with podman but not with docker. However we let the task run
-        # anyway, since all clients will look with a HEAD request before attemting to upload a blob
+        # This mechanism seems to work with podman but not with docker. However, we let the task run
+        # anyway, since all clients will look with a HEAD request before attempting to upload a blob
         # again.
         try:
             upload = models.Upload.objects.get(pk=pk, repository=repository)
         except models.Upload.DoesNotExist as e_upload:
-            # Upload has been deleted => task has started or even finished
-            task = Task.objects.filter(
-                name__endswith="add_and_remove",
-                reserved_resources_record__contains=[f"shared:upload:{pk}"],
-            ).last()
-            if not task:
-                # No upload and no task for it => the upload probably never existed
-                # return 404
+            try:
+                blob = repository.pending_blobs.get(digest=digest)
```

Comment:
i wonder if we should also look for the blob in the latest repo version. what about a situation when there 2 concurrent pushes of the same content and this line of code will hit in first push right when second push is removing blob from pending and adding those to repo?

Reply:
Absolutely and we should do so first, i think.